src/error-snippets/index.ts

```diff
@@ -0,0 +1,36 @@
+import { findRelevantStackFrame, resolveWithStackFrame } from "./frames";
+import { extractSourceMaps, resolveWithSourceMap } from "./source-maps";
+import { getSourceCodeFile, formatErrorSnippet } from "./utils";
+import logger from "../utils/logger";
+import type { ResolvedFrame, SufficientStackFrame, WithSnippetError } from "./types";
+
+const stackFrameResolver = async (stackFrame: SufficientStackFrame): Promise<ResolvedFrame> => {
+    const fileContents = await getSourceCodeFile(stackFrame.fileName);
+    const sourceMaps = await extractSourceMaps(fileContents, stackFrame.fileName);
+
+    return sourceMaps ? resolveWithSourceMap(stackFrame, sourceMaps) : resolveWithStackFrame(stackFrame, fileContents);
+};
+
+export const extendWithCodeSnippet = async (err: WithSnippetError): Promise<WithSnippetError> => {
+    if (!err) {
+        return err;
```

Comment:
I totally understand what you're doing here, but it looks weird. why not explicitly return something like `null` or `undefined`?

Reply:
Why do you want to explicitly return `null` or `undefined` here?

I want it to return the same `Object.Is` equality object, and dont think about "what if it returned null" in other places. Its easier to think like "extendWithCodeSnippet added some code snippet if it could. If it didn't, anyway we won't try to do something with it"